docs/redux.md

```diff
@@ -0,0 +1,118 @@
+you can see from there what props are available for each component when they are rendered.
+
+Naturally, there is a need for higher level components which fetch new data and define what props
+are passed down to presentational components. In Redux terminology, those components are called
+_Containers_. FTW has defined all the containers inside a folder called `src/containers/`. It
+contains all the pages and a special container for top bar (TopbarContainer) since that is a very
+complex component and it is shared with almost every page.
+
+The actual container setup of a page level component can be found from the bottom of the component
+file. For example, `src/containers/TransactionPage/TransactionPage.js` connects itself to Redux
+store with `mapStateToProps` and `mapDispatchToProps` functions:
+
+```js
+const TransactionPage = compose(connect(mapStateToProps, mapDispatchToProps), injectIntl)(
+  TransactionPageComponent
+);
+```
+
+## Duck files
+
+Inside `src/containers/<ComponentName>` folder, we have also a special `<ComponentName>.duck.js`
+file. These files wrap all the other page-specific Redux concepts into a single file. Instead of
+writing action types, action creators and reducers in separate files (and spread them around our
+folder structure), FTW tries to keep pages isolated. Page specific actions, store updates, and data
+fetches happen inside their respective folder - just look for a file which name follows a pattern:
+`<ComponentName>.duck.js`. This pattern was first proposed by Erik Rasmussen and you can read more
+about it from [his repository](https://github.com/erikras/ducks-modular-redux).
+
+### Global reducers
+
+Some reducers are needed in several pages. These global reducers we have defined inside `src/ducks/`
+folder with their respective `*.duck.js` files. Most important global duck files are `user.duck.js`
+and `marketplaceData.duck.js`.
+
+## Setting up Redux
+
+Container specific reducers are gathered and exported inside `src/containers/reducers.js` file and
+global reducers are exported respectively in a file `src/ducks/index.js`.
+
+With those exports, we are able to create appReducer (`src/reducers.js`):
+
+```js
+import { combineReducers } from 'redux';
+import * as globalReducers from './ducks';
+import * as pageReducers from './containers/reducers';
+
+const appReducer = combineReducers({ ...globalReducers, ...pageReducers });
+```
+
+`appReducer` is called by `createReducer` function, which is called inside of `configureStore`
+function (in `src/store.js`).
+
+This setup creates a store structure that separates container specific state as well as global data
+by their reducer names. Together with Ducks module naming schema, this means that:
+
+* the state of the `ListingPage` can be found from `state.ListingPage` and
+* the state of the global `user` object can be found from `state.user`.
+
+## Advanced Redux concepts: thunks
```

Comment:
Not really "advanced", but ok. :P

Reply:
It is advanced since Redux docs say it's advanced :P